geniusyield-market-maker/src/GeniusYield/MarketMaker/Prices/Taptools.hs

```diff
@@ -70,7 +71,19 @@ data TaptoolsPriceException
   = TaptoolsError !Text               -- ^ Other taptools error.
   | TaptoolsClientError !ClientError  -- ^ Servant client error while querying taptools.
   deriving stock (Eq, Show)
-  deriving anyclass Exception
+
+instance Exception TaptoolsPriceException where
+  displayException (TaptoolsError msg) = "Taptools Error: " ++ unpack msg
+  displayException (TaptoolsClientError err) = "Taptools Client Error: " ++ displayException (sanitizeClientFail err)
```

Comment:
Likewise, instead of modifying `displayException` one can make sure the exception raised is _correct_ by construction. Here _correct_ refers to api key being concealed, like [here](https://github.com/geniusyield/dex-contracts-api/blob/4ff939875bada81e47ff0068669e9719b219943c/geniusyield-server-lib/src/GeniusYield/Server/Dex/HistoricalPrices/TapTools/Client.hs#L139-L140).

Reply:
You are right in that handling an exception "correct by construction" is cleanest.  But here I argue that the method of modifying `displayException` works because the only place where logging of Taptools error occurs is in function [logPricesProviderFail](https://github.com/geniusyield/market-maker/blob/fda693e5ee990fceea005e1a7a31829273f84b15/geniusyield-market-maker/src/GeniusYield/MarketMaker/Prices.hs#L566), and there `displayException` is used.

Nevertheless I modified this instance definition to incorporate the `FailureResponse` case, following the way it is done [here](https://github.com/geniusyield/dex-contracts-api/blob/4ff939875bada81e47ff0068669e9719b219943c/geniusyield-server-lib/src/GeniusYield/Server/Utils.hs#L55) with `hideServantClientErrorHeader`.